std/assembly/array.ts

```diff
@@ -105,9 +105,14 @@ export class Array<T> extends ArrayBufferView {
   }
 
   @operator("[]=") private __set(index: i32, value: T): void {
-    ensureSize(changetype<usize>(this), index + 1, alignof<T>());
-    this.__unchecked_set(index, value);
-    if (index >= this.length_) this.length_ = index + 1;
+    if (<u32>index >= <u32>this.length_) {
+      if (index < 0) throw new RangeError(E_INDEXOUTOFRANGE);
+      ensureSize(changetype<usize>(this), index + 1, alignof<T>());
+      this.__unchecked_set(index, value);
+      this.length_ = index + 1;
+    } else {
+      this.__unchecked_set(index, value);
```

Comment:
Should even be possible to pull this and the previous call out to after the `if-else` :)

Reply:
done!